docs/tasks/administer-cluster/kms-provider.md

```diff
@@ -0,0 +1,182 @@
+---
+approvers:
+- smarterclayton
+title: Using a KMS provider for data encryption
+---
+{% capture overview %}
+This page shows how to configure a Key Management Service (KMS) provider and plugin to enable secret data encryption.
+{% endcapture %}
+
+{% capture prerequisites %}
+
+* {% include task-tutorial-prereqs.md %}
+
+* Kubernetes version 1.10.0 or later is required
+
+* etcd v3 or later is required
+
+*The KMS provider is alpha in Kubernetes version 1.10.0, which means that it may change without notice. You may be required to decrypt your data prior to upgrading to 1.11.0
+
+{% endcapture %}
+
+{% capture steps %}
+
+The KMS encryption provider uses an envelope encryption scheme to encrypt data in etcd. The Key encryption keys (KEKs) are 
```

Comment:
Please explain what is a "KMS encryption provider" and what is a "KMS encryption plugin" here, before diving into the configuration steps.

Reply:
have added a few more details.